scripts/dev/model_introspection_221b_utils.py

```
@@ -0,0 +1,61 @@
+    url: str,
+    *,
+    method: str = "GET",
+    payload: dict[str, Any] | None = None,
+    timeout_sec: float = 30.0,
+) -> tuple[int, Any]:
+    data_bytes: bytes | None = None
+    headers: dict[str, str] = {}
+    if payload is not None:
+        data_bytes = json.dumps(payload).encode("utf-8")
+        headers["Content-Type"] = "application/json"
+    req = request.Request(url=url, method=method, data=data_bytes, headers=headers)
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)